### PR TITLE
ZIO-GCP: cloud storage

### PR DESCRIPTION
**Pull request by mikail-khan**
#29 & #56 - cloud storage service api

### CODE REVIEW

**Comment by mikail-khan** [comment]
@mijicd - will implement the service definition over the course of next week depending on how busy work gets.

File: modules/firestore/src/main/scala/zio/gcp/firestore/Firestore.scala [hunk 1]
```diff
@@ -27,7 +27,8 @@ object Firestore {
 
   def live: Layer[Throwable, Firestore] =
     ZLayer.fromManaged {
-      val acquire = IO.effect(FirestoreOptions.getDefaultInstance().toBuilder().build().getService())
+      val acquire: Task[firestore.Firestore] =
+        IO.effect(FirestoreOptions.getDefaultInstance().toBuilder().build().getService())
```
Comment (mijicd) [inline modules/firestore/src/main/scala/zio/gcp/firestore/Firestore.scala:31]:
```suggestion
      val acquire = Task.effect(FirestoreOptions.getDefaultInstance().toBuilder().build().getService())
```

File: modules/storage/src/main/scala/zio/gcp/storage/Storage.scala [hunk 2]
```diff
@@ -1,13 +1,307 @@
...
-trait Storage {
-  val storage: Storage.Service[Any]
-}
+import scala.jdk.CollectionConverters._
 
 object Storage {
-  trait Service[R] {
-    def create(bucketInfo: BucketInfo): URIO[R, BucketInfo]
+  trait Service {
+    def batch(): Task[StorageBatch]
```
Comment (mijicd) [inline modules/storage/src/main/scala/zio/gcp/storage/Storage.scala:17]: No need for `()` as it's a `Task`, not the side-effecting one.

File: modules/storage/src/main/scala/zio/gcp/storage/Storage.scala [hunk 3]
```diff
@@ -1,13 +1,307 @@
...
 
-trait Storage {
-  val storage: Storage.Service[Any]
-}
+import scala.jdk.CollectionConverters._
 
 object Storage {
-  trait Service[R] {
-    def create(bucketInfo: BucketInfo): URIO[R, BucketInfo]
+  trait Service {
```
Comment (mijicd) [inline modules/storage/src/main/scala/zio/gcp/storage/Storage.scala:16]: Are all these variants needed? If not, I would propose exposing only a couple of most-frequently used, and provide a "hook", to execute the other ones safely. I believe `zio-ftp` has a good example how to do this.

cc @regis-leray

Comment (mikail-khan) [reply modules/storage/src/main/scala/zio/gcp/storage/Storage.scala:16]: I looked at the underlying implementation of the Java library and have removed the ones which superfluous.

File: modules/firestore/src/main/scala/zio/gcp/firestore/Firestore.scala (hunk 1 above)
Comment (mikail-khan) [reply modules/firestore/src/main/scala/zio/gcp/firestore/Firestore.scala]: fixed

File: modules/storage/src/main/scala/zio/gcp/storage/Storage.scala (hunk 2 above)
Comment (mikail-khan) [reply modules/storage/src/main/scala/zio/gcp/storage/Storage.scala]: fixed

**Pull request merged**